src/utils.ts

```diff
@@ -54,11 +54,23 @@ export const getProxyEnv = (requestURL: URL): string => {
 
   // get proxy based on request url's protocol
   if (requestURL.protocol == "http:") {
-    return process.env.HTTP_PROXY || process.env.http_proxy || null;
+    return (
+      process.env.HTTP_PROXY ||
+      process.env.http_proxy ||
+      process.env.HTTPS_PROXY ||
+      process.env.https_proxy ||
+      null
```

Comment:
let's stick with request implementation https://github.com/request/request/blob/3c0cddc7c8eb60b470e9519da85896ed7ee0081e/lib/getProxyFromURI.js#L62. i,e no fallback on https for http url

given that it's a battle-tested module

Reply:
I was thinking the same, stand by...